### PR TITLE
Fix/handle done menu visibility and crop error

### PR DESCRIPTION
**Pull request by ashiagr**
This PR handles done menu visibility and crop error as part of https://github.com/wordpress-mobile/WordPress-Android/pull/11392.

Dependent PR: https://github.com/wordpress-mobile/WordPress-Android/pull/11447 to add `Event` wrapper class.

TODO
- [ ] Add "Needs Copy Review" label for error message
`<string name="error_failed_to_crop_and_save_image">Failed to crop and save image, please try again.</string>`

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt [hunk 1]
```diff
@@ -75,22 +102,36 @@ class CropFragment : Fragment(), UCropFragmentCallback {
     }
 
     private fun showThirdPartyCropFragmentWithBundle(bundle: Bundle) {
-        thirdPartyCropFragment = UCropFragment.newInstance(bundle)
-        childFragmentManager.beginTransaction()
+        var thirdPartyCropFragment = childFragmentManager
+                .findFragmentByTag(UCropFragment.TAG) as? UCropFragment
+
+        if (thirdPartyCropFragment == null || !thirdPartyCropFragment.isAdded) {
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt]: > did you add the code just to be safe?
Added checks just to be safe

File: ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt [hunk 2]
```diff
@@ -48,12 +53,33 @@ class CropFragment : Fragment(), UCropFragmentCallback {
...
 
     private fun setupObservers() {
-        viewModel.showCropScreenWithBundle.observe(this, Observer { cropOptionsBundleWithFilesInfo ->
-            showThirdPartyCropFragmentWithBundle(cropOptionsBundleWithFilesInfo)
+        viewModel.uiState.observe(this, Observer { uiState ->
+            when (uiState) {
+                is UiStartLoadingWithBundleState -> {
+                    showThirdPartyCropFragmentWithBundle(uiState.bundle)
+                }
+                else -> { // Do nothing
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt]: Added here 03488de

File: ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt [hunk 3]
```diff
@@ -48,12 +53,33 @@ class CropFragment : Fragment(), UCropFragmentCallback {
...
+        viewModel.cropAndSaveImageState.observe(this, Observer { state ->
+            when (state) {
+                is ImageCropAndSaveStartState -> {
+                    val thirdPartyCropFragment = childFragmentManager
+                            .findFragmentByTag(UCropFragment.TAG) as? UCropFragment
+                    thirdPartyCropFragment?.let {
+                        if (thirdPartyCropFragment.isAdded) {
+                            thirdPartyCropFragment.cropAndSaveImage()
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt]: Added here 87d1dd1

File: ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (malinajirka) [reply ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt]: 👍 In general, I think it's better to let the app crash when something unexpected happens. But I guess it's ok in this case to introduce the check. Thanks for the clarification!

**Pull request merged**